app/Models/Beatmapset.php

```diff
@@ -646,6 +621,56 @@ public function scanBMForBG($beatmapFilename)
         return $imageFilename;
     }
 
+    public function disqualify(User $user, $comment)
+    {
+        if (!$this->isQualified()) {
+            return false;
+        }
+
+        DB::transaction(function () use ($user, $comment) {
+            $this->events()->create(['type' => BeatmapsetEvent::DISQUALIFY, 'user_id' => $user->user_id, 'comment' => $comment]);
+            $this->approved = 0;
+            $this->save();
+        });
+
+        return true;
+    }
+
+    public function qualify()
+    {
+        if (!$this->isPending()) {
+            return false;
+        }
+
+        DB::transaction(function () {
+            $this->events()->create(['type' => BeatmapsetEvent::QUALIFY]);
+            $this->approved = 3;
```

Comment:
cute number here :100: 

I think it's better to flip the `STATES` constant so the usage here and somewhere else is `static::STATES['qualified']` and the other usage (in hopefully single function) is `array_search_null(...stuff...)`.